vic/drivers/shared_image/src/get_global_domain.c

```diff
@@ -103,17 +118,77 @@ get_global_domain(char          *nc_name,
         }
     }
 
+    // allocate memory for variables
+    var = malloc(global_domain->ncells_total * sizeof(*var));
+    check_alloc_status(var, "Memory allocation error.");
+
+    // get area
+    // TBD: read var id from file
+    get_nc_field_double(domain_nc_name, global_domain->info.area_var,
+                        d2start, d2count, var);
+    for (i = 0; i < global_domain->ncells_total; i++) {
+        global_domain->locations[i].area = var[i];
+    }
+
+    // get fraction
+    // TBD: read var id from file
+    get_nc_field_double(domain_nc_name, global_domain->info.frac_var,
+                        d2start, d2count, var);
+    for (i = 0; i < global_domain->ncells_total; i++) {
+        global_domain->locations[i].frac = var[i];
+    }
+
+    // get lat and lon coordinates
+    get_nc_latlon(domain_nc_name, global_domain, global_domain);
+
+    // check whether lat and lon coordinates in the parameter file match those
+    // in the domain file
+    compare_ncdomain_with_global_domain(param_nc_name);
+
+    // free memory
+    free(var);
+    free(run);
+
+    return global_domain->ncells_active;
+}
+
+
+/******************************************************************************
+ * @brief    Get lat and lon coordinates information from a netCDF file and
+             store in nc_domain structure
+ *****************************************************************************/
+void
+get_nc_latlon(char          *nc_name,
+              domain_struct *global_domain,
+              domain_struct *nc_domain)
```

Comment:
I find this confusing as well. It makes the function do more than it needs to anyway. If you are sure that the domain information from `global_domain` can simply be copied to `nc_domain`, then do so explicitly before you enter the function. For example, write a small `copy_domain(domain_struct *domain_from, domain_struct *domain_to)` function. Call that first and then call `get_nc_latlon(char *nc_name, domain_struct *nc_domain)`. That way there is no hidden beahvior (the copy that happens as a side effect in `get_bc_latlon()`.